core/templates/pages/feedback-updates-page/feedback-updates-page.component.spec.ts

```diff
@@ -0,0 +1,812 @@
+      // getValueOfSubscriptionSortKey is used in a *ngFor directive.
+      // Note that given subscription list is not sorted.
+      const expectedInnerText = ['Biology', 'Algebra', 'Three Balls', 'Zebra'];
+      feedbackListNameNodes.forEach((titleNode, index) => {
+        expect(titleNode.innerText).toContain(expectedInnerText[index]);
+      });
+    }));
+
+    it('should sort feedback updates given sorting property as last updated' +
+      ' in descending order', fakeAsync(() => {
+      // The default sort option for Feedback Updates is last updated.
+      expect(component.currentFeedbackThreadsSortType)
+        .toBe('lastUpdatedMsecs');
+      expect(component.isCurrentFeedbackSortDescending).toBeTrue();
+      expect(component.getValueOfFeedbackThreadSortKey())
+        .toBe('lastUpdatedMsecs');
+
+      component.setFeedbackSortingOptions('lastUpdatedMsecs');
+
+      expect(component.isCurrentFeedbackSortDescending).toBeFalse();
+
+      tick();
+      fixture.detectChanges();
+
+      const feedbackListNameNodes =
+        fixture.debugElement.nativeElement
+          .querySelectorAll('.e2e-test-feedback-exploration') as
+           NodeListOf<HTMLElement>;
+
+      // The forEach loop is being used here because
+      // getValueOfSubscriptionSortKey is used in a *ngFor directive.
+      // Note that given subscription list is not sorted.
+      const expectedInnerText = ['Zebra', 'Three Balls', 'Algebra', 'Biology'];
+      feedbackListNameNodes.forEach((titleNode, index) => {
+        expect(titleNode.innerText).toContain(expectedInnerText[index]);
+      });
+    }));
+
+    it('should sort feedback updates given sorting property as exploration' +
+      ' in ascending order', fakeAsync(() => {
+      // The default sort option for Feedback Updates is last updated.
+      expect(component.currentFeedbackThreadsSortType)
+        .toBe('lastUpdatedMsecs');
+      expect(component.isCurrentFeedbackSortDescending).toBeTrue();
+      expect(component.getValueOfFeedbackThreadSortKey())
+        .toBe('lastUpdatedMsecs');
+
+      component.setFeedbackSortingOptions('explorationTitle');
+
+      expect(component.currentFeedbackThreadsSortType)
+        .toBe('explorationTitle');
+      expect(component.getValueOfFeedbackThreadSortKey())
+        .toBe('explorationTitle');
+
+      tick();
+      fixture.detectChanges();
+
+      const feedbackListNameNodes =
+        fixture.debugElement.nativeElement
+          .querySelectorAll('.e2e-test-feedback-exploration') as
+           NodeListOf<HTMLElement>;
+
+      // The forEach loop is being used here because
+      // getValueOfSubscriptionSortKey is used in a *ngFor directive.
+      // Note that given subscription list is not sorted.
+      const expectedInnerText = ['Algebra', 'Biology', 'Three Balls', 'zebra'];
+      feedbackListNameNodes.forEach((titleNode, index) => {
+        expect(titleNode.innerText).toContain(expectedInnerText[index]);
+      });
```

Comment:
Sorry if I was not clear. You need to append the `titleNode.innerText` into a new list. After which, you will need to use `toBe` to compare the lists. This is to ensure that the 2 lists are exactly the same and that there are no additional elements which can happen in this tests case.

Reply:
```
 const expectedInnerText = ['Algebra', 'Biology', 'Three Balls', 'zebra'];
 let titleNodeNames: string[ ] = [ ]
      feedbackListNameNodes.forEach((titleNode) => {
        titleNodeNames.push(titleNode.innerText);
      });
 expect (titleNodeNames).toEqual(expectedInnerText);
```

I have tried this but it gives an error

![image](https://github.com/oppia/oppia/assets/100484672/7d888adc-14d4-43f4-868c-635d11647bd4)